.github/workflows/main.yml

```
@@ -5,12 +5,41 @@ on:
+  update_version:
+    if: "!github.event.release.prerelease"  # Only run for stable releases
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Required for version comparison
+
+      - name: Extract version
+        id: version
+        run: echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_OUTPUT
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:45: Double quote to prevent globbing and word splitting

(shellcheck)